interp/src/interpreter/interpret_group.rs

```diff
@@ -165,69 +160,90 @@ fn interp_assignments<'a, I: Iterator<Item = &'a ir::Assignment>>(
+                //value driving the port
+                assigned_ports.insert(get_const_from_rrc(&assignment.dst));
+                //ok now proceed
                 let old_val = working_env.get(&assignment.dst.borrow());
-                let new_val = working_env.get_as_val(&assignment.src.borrow());
+                let new_val_ref =
+                    working_env.get_as_val(&assignment.src.borrow());
 
                 // no need to make updates if the value has not changed
-                if old_val != new_val.into() {
-                    val_changed_flag = true;
-                    updates_list.push(Rc::clone(&assignment.dst));
-
-                    // Using a TLV to simulate updates happening after reads
-                    // Note: this is a hack and should be removed pending
-                    // changes in the environment structures
-                    // see: https://github.com/cucapra/calyx/issues/549
+                let port = assignment.dst.clone(); // Rc clone
+                let new_val: OutputValue = new_val_ref.clone().into();
+                updates_list.push((port, new_val));
 
-                    let tmp_old = match old_val.clone_referenced() {
-                        OutputValue::ImmediateValue(iv) => Some(iv),
-                        OutputValue::LockedValue(tlv) => tlv.old_value,
-                        OutputValue::PulseValue(pv) => Some(pv.take_val()),
-                    };
-
-                    let new_val = OutputValue::LockedValue(
-                        TimeLockedValue::new(new_val.clone(), 0, tmp_old),
-                    );
-
-                    let port = &assignment.dst.borrow();
-
-                    working_env.update_val(&port, new_val);
+                if old_val != new_val_ref.into() {
+                    val_changed_flag = true;
                 }
             }
         }
 
-        // Remove the placeholder TLVs
-        for port in updates_list {
-            if let Entry::Occupied(entry) = working_env.entry(&port.borrow()) {
-                let mut_ref = entry.into_mut();
-                let v = std::mem::take(mut_ref);
-
-                *mut_ref = if v.is_tlv() {
-                    v.unwrap_tlv().unlock().into()
-                } else {
-                    // this branch should be impossible since the list of
-                    // ports we're iterating over are only those w/ updates
-                    unreachable!()
-                }
+        //now assign rest to 0
+        //first get all that need to be 0
+        for port in &possible_ports - &assigned_ports {
+            //need to set to zero, because unassigned
+            //ok now proceed
+
+            //need to find appropriate-sized 0, so just read
+            //width of old_val
+
+            let old_val = working_env.get_as_val_const(port);
+            let old_val_width = old_val.width(); //&assignment.dst.borrow().width()
```

Comment:
Could this just be `port.width`? (Looks like [Port](https://github.com/cucapra/calyx/blob/d4e3789861af7edc810656601faf4cad91119c21/calyx/src/ir/structure.rs#L38) has that information.)

Reply:
In this case `port` is of type `* const Port` so it can't be used for anything other than indexing the maps w/o use of unsafe

Reply:
Got it; good point.